examples/remote-settings-cli/src/downloader.rs

```diff
@@ -0,0 +1,289 @@
+use anyhow::{Context, Result};
+use futures::{stream::FuturesUnordered, StreamExt};
+use indicatif::{MultiProgress, ProgressBar, ProgressStyle};
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+use std::{collections::HashMap, path::PathBuf, sync::Arc};
+
+use crate::git_ops::GitOps;
+
+const REMOTE_SETTINGS_URL: &str = "https://firefox.settings.services.mozilla.com/v1";
+const DEFAULT_OUTPUT_DIR: &str = "components/remote_settings";
+const COLLECTIONS: &[&str] = &[
+    "anti-tracking-url-decoration",
+    "cookie-banner-rules-list",
+    "devtools-compatibility-browsers",
+    "devtools-devices",
+    "hijack-blocklists",
+    "language-dictionaries",
+    "password-recipes",
+    "password-rules",
+    "search-config-icons",
+    "search-config-v2",
+    "search-config-overrides-v2",
+    "search-telemetry-v2",
+    "sites-classification",
+    "top-sites",
+    "translations-models",
+    "translations-wasm",
+    "url-classifier-skip-urls",
+    "url-parser-default-unknown-schemes-interventions",
+    "urlbar-persisted-search-terms",
+    "websites-with-shared-credential-backends",
+];
```

Comment:
Instead of having a hardcoded list, on desktop we have a folder and we use the file presents in that folder as the list to fetch.
Touching an empty and running the cargo command would update it.

Reply:
Can you explain "touching an empty and running the cargo command"? What happens if the files don't previously exist? It's probably not going to happen because than we cannot build, but I would like these collection names at least somewhere referred to in the codebase.

Reply:
I removed the list and simply download all available collections. This might be too much. We can also manually download them, and just update them based on file name.

Reply:
> Can you explain "touching an empty and running the cargo command"?

Sorry if it was not clear.

My idea was, in order to package a new collection, to do this:

```
$ touch components/remote-settings/dumps/main/password-rules.json
$ cargo remote-settings download --create-pr
```

> I removed the list and simply download all available collections. This might be too much. 

Yes, maybe, unless the developers/packagers can pick at build time which collections to include. But that would be equivalent to having a hardcoded list.

> We can also manually download them, and just update them based on file name.

The idea on desktop is that we use the list of files in `services/settings/dumps/*/*.json` to determine which collections to download